drivers/video/video_mcux_csi.c

```diff
@@ -132,9 +165,17 @@ static int video_mcux_csi_set_fmt(const struct device *dev,
 	data->pixelformat = fmt->pixelformat;
 	data->csi_config.bytesPerPixel = bpp;
 	data->csi_config.linePitch_Bytes = fmt->pitch;
+#if defined(CONFIG_VIDEO_MCUX_MIPI_CSI2RX)
+	if (fmt->pixelformat != VIDEO_PIX_FMT_XRGB32 && fmt->pixelformat != VIDEO_PIX_FMT_XYUV32) {
+		return -ENOTSUP;
+	}
+	video_pix_fmt_convert(&format, false);
+	data->csi_config.dataBus = kCSI_DataBus24Bit;
```

Comment:
Maybe instead of making this dependent on Kconfig we should move the `dataBus` configuration to devicetree? Not sure if all cameras are 8bit bus width, or if some would use more pins

Reply:
We can't make this configurable in devicetree. The databus is either 8 bits or 24 bits depending on the path of image data. If it passes by mipi csi2rx, data are implicitly converted to 32-bit format and 24 bits need to be set for it to work.